### PR TITLE
refine(qrm): calculate pool size with much accuracy

### PR DESCRIPTION
**Pull request by csfldf**
refine(qrm): calculate pool size with much accuracy

### CODE REVIEW

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_allocation_handlers.go [hunk 1]
```diff
@@ -911,20 +922,37 @@ func (p *DynamicPolicy) generatePoolsAndIsolation(poolsQuantityMap map[string]in
 		proportionalPoolsQuantityMap := make(map[string]int)
 
 		for poolName, poolQuantity := range poolsQuantityMap {
```
Comment (waynepeking348) [inline pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_allocation_handlers.go:924]: can those logic be refined (for instance, put in another function)? kind of too large to understand ...

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/state/util.go [hunk 2]
```diff
@@ -146,10 +149,16 @@ func GetSharedQuantityMapFromPodEntries(podEntries PodEntries, ignoreAllocationI
 			}
 
 			if poolName := allocationInfo.GetOwnerPoolName(); poolName != advisorapi.EmptyOwnerPoolName {
-				ret[poolName] += GetContainerRequestedCores()(allocationInfo)
+				preciseQuantityMap[poolName] += GetContainerRequestedCores()(allocationInfo)
 			}
 		}
 	}
+
+	for poolName, preciseQuantity := range preciseQuantityMap {
```
Comment (waynepeking348) [inline pkg/agent/qrm-plugins/cpu/dynamicpolicy/state/util.go:157]: will it exceeds node total if all pools are ceiled?

Comment (csfldf) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/state/util.go:157]: yes, the sum of ret will exceed node total here. but the generatePoolsAndIsolation will adjust to guarantee cpuset of every pool won't exceed current available cpus.

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_allocation_handlers.go [hunk 3]
```diff
@@ -911,20 +922,37 @@ func (p *DynamicPolicy) generatePoolsAndIsolation(poolsQuantityMap map[string]in
...
-			proportionalSize := general.Max(getProportionalSize(poolQuantity, poolsTotalQuantity, availableSize), 1)
+			proportionalSize := general.Max(getProportionalSize(poolQuantity, poolsTotalQuantity, availableSize, true /*ceil*/), 1)
```
Comment (luomingmeng) [inline pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_allocation_handlers.go:925]: the comment `ceil` means `true` will ceil the result, and the `false` is not?

Comment (csfldf) [reply pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_allocation_handlers.go:925]: yes

**Review by waynepeking348 — APPROVED** [review]

**Review by luomingmeng — APPROVED** [review]

**Pull request merged**